config/config_test.go

```diff
@@ -0,0 +1,205 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	validFile = `
+{
+  "Servers": {
+    "rancherDefault": {
+      "accessKey": "the-access-key",
+      "secretKey": "the-secret-key",
+      "tokenKey": "the-token-key",
+      "url": "https://example.com",
+      "project": "cluster-id:project-id",
+      "cacert": "",
+      "kubeCredentials": null,
+      "kubeConfigs": null
+    }
+  },
+  "CurrentServer": "rancherDefault"
+}`
+	invalidFile = `invalid config file`
+)
+
+func Test_GetFilePermissionWarnings(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		mode             os.FileMode
+		expectedWarnings int
+	}{
+		{
+			name:             "neither group-readable nor world-readable",
+			mode:             os.FileMode(0600),
+			expectedWarnings: 0,
+		},
+		{
+			name:             "group-readable and world-readable",
+			mode:             os.FileMode(0644),
+			expectedWarnings: 2,
+		},
+		{
+			name:             "group-readable",
+			mode:             os.FileMode(0640),
+			expectedWarnings: 1,
+		},
+		{
+			name:             "world-readable",
+			mode:             os.FileMode(0604),
+			expectedWarnings: 1,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			assert := assert.New(t)
+
+			dir, err := os.MkdirTemp("", "rancher-cli-test-*")
+			assert.NoError(err)
+			defer os.RemoveAll(dir)
+
+			path := filepath.Join(dir, "cli2.json")
+			err = os.WriteFile(path, []byte(validFile), tt.mode)
+			assert.NoError(err)
+
+			warnings, err := GetFilePermissionWarnings(path)
+			assert.NoError(err)
+			assert.Len(warnings, tt.expectedWarnings)
+		})
+	}
+}
+
+func Test_Permission(t *testing.T) {
+	t.Parallel()
+
+	// New config files should have 0600 permissions
+	t.Run("new config file", func(t *testing.T) {
+		t.Parallel()
```

Comment:
I don't think that this one and the `existing config file` can be marked as parallel since they use the same underlying file name. While you can keep the Parallel marker for the permission function (on l82), I don't think that this one (and the one in L111) is safe.

Reply:
The `cli2.json` file is created in different folders since each subtest calls `MkdirTemp` so I believe it's safe. Wdyt?